src/mesh/gmsh_utils.jl

```diff
@@ -387,10 +392,12 @@ function gen_rectangle_mesh(
     surf = gmsh.model.geo.addPlaneSurface([ABCD])
 
     # Mesh settings
-    gmsh.model.geo.mesh.setTransfiniteCurve(AB, nx)
-    gmsh.model.geo.mesh.setTransfiniteCurve(BC, ny)
-    gmsh.model.geo.mesh.setTransfiniteCurve(CD, nx)
-    gmsh.model.geo.mesh.setTransfiniteCurve(DA, ny)
+    if transfinite_lines
+        gmsh.model.geo.mesh.setTransfiniteCurve(AB, nx)
+        gmsh.model.geo.mesh.setTransfiniteCurve(BC, ny)
+        gmsh.model.geo.mesh.setTransfiniteCurve(CD, nx)
+        gmsh.model.geo.mesh.setTransfiniteCurve(DA, ny)
+    end
 
     (transfinite || type == :quad) && gmsh.model.geo.mesh.setTransfiniteSurface(surf)
```

Comment:
we should rename `transfinite` to `transfinite_surface` for more coherence with `transfinite_lines`